src/history.rs

```diff
@@ -804,34 +809,38 @@ impl HistoryImpl {
         // After locking it, we need to stat the file at the path; if there is a new file there, it
         // means the file was replaced and we have to try again.
         // Limit our max tries so we don't do this forever.
-        let mut history_fd = AutoCloseFd::new(-1);
+        let mut history_fd = None;
         for _i in 0..MAX_SAVE_TRIES {
-            let fd = AutoCloseFd::new(wopen_cloexec(&history_path, O_WRONLY | O_APPEND, 0));
-            if !fd.is_valid() {
+            let Ok(fd) = wopen_cloexec(
+                &history_path,
+                OFlag::O_WRONLY | OFlag::O_APPEND,
+                Mode::empty(),
+            ) else {
                 // can't open, we're hosed
                 break;
-            }
+            };
+
             // Exclusive lock on the entire file. This is released when we close the file (below). This
             // may fail on (e.g.) lockless NFS. If so, proceed as if it did not fail; the risk is that
             // we may get interleaved history items, which is considered better than no history, or
             // forcing everything through the slow copy-move mode. We try to minimize this possibility
             // by writing with O_APPEND.
             unsafe {
-                Self::maybe_lock_file(fd.fd(), LOCK_EX);
+                Self::maybe_lock_file(fd.as_raw_fd(), LOCK_EX);
```

Comment:
Can we change this function to take a reference to an owned fd?

Reply:
Indeed `maybe_lock_file` is no longer called with raw_fd anywhere, replaced with generic `AsFd`. Should work with `OwnedFd`, ` BorrowedFd` and `File`